wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wdl/WdlWriterImpl.scala

```diff
@@ -294,17 +294,15 @@ object WdlWriterImpl {
 
   implicit val placeholderAttributeSetWriter: WdlWriter[PlaceholderAttributeSet] = new WdlWriter[PlaceholderAttributeSet] {
     override def toWdlV1(a: PlaceholderAttributeSet): String = {
-      Map(
+      val attrStrings = Map(
         "sep" -> a.sepAttribute,
         "true" -> a.trueAttribute,
         "false" -> a.falseAttribute,
         "default" -> a.defaultAttribute
-      ).map({ case (attrKey: String, maybeValue: Option[String]) =>
-        maybeValue match {
-          case Some(value) => attrKey + "=\"" + value + "\""
-          case None => ""
-        }
-      }).filterNot(_.length == 0).mkString(sep = " ")
+      ).collect({ case (attrKey: String, Some(value)) => attrKey + "=\"" + value + "\"" })
+        .filterNot(_.length == 0)
+
+      if (attrStrings.isEmpty) "" else attrStrings.mkString(start = "", sep = " ", end = " ")
```

Comment:
I ended up refactoring this so that I could get this extra `" "` at the end, which turned out to be important in our call caching algorithm (but only because we have to lay it out the same as the draft 2 printer did. Spaces between attributes shouldn't matter in the original source 🤞)